Source/Result.swift

```diff
@@ -100,3 +100,82 @@ extension Result: CustomDebugStringConvertible {
         }
     }
 }
+
+// MARK: - Functional Tools
+
+extension Result {
+    /// Returns the success value, or throws the failure error.
+    ///
+    ///     let possibleString: Result<String> = .success("success")
+    ///     try print(possibleString.unwrap())
+    ///     // Prints "success"
+    ///
+    ///     let noString: Result<String> = .failure(error)
+    ///     try print(noString.unwrap())
+    ///     // Throws error
+    func unwrap() throws -> Value {
```

Comment:
I was waiting for this question. Yes, I did not choose `dematerialize`. I don't think Alamofire needs/wants to use bleeding edge functional concepts or vocabulary.

`dematerialize` is long. `dematerialize` is spectacular. `dematerialize` is abstract. `dematerialize` is pedantic. `dematerialize` contains a latent implicit disagreement on swift's choice of try/catch/throws instead of a "proper sum type" that needs to be "materialized". `dematerialize` is not yet a term of art.

On the other side, Rust's `unwrap` sounds just as boring as it needs to be. It fits better in the multi-paradigms Swift language. It fits better with some Swift evolution discussions around wrapper types such as Optional and other various boxes. Unfortunately, this mailing list is a chore when you look for something in it, but I'd like to put my finger on a proposal/post by @erica - maybe she'll read this and help ;-)